KinUtil/KinUtil/source/Promise.swift

```diff
@@ -174,7 +174,7 @@ extension Promise {
 
 extension Promise {
     private func run(_ block: () -> (), on queue: DispatchQueue?) {
-        if let queue = queue {
+        if let queue = queue, queue != .main {
```

Comment:
Are you sure this is necessary?  `sync()` is supposed to be a no-op when the target queue is also the source queue.  That is, the block should just be executed directly by the API, and not scheduled.

Reply:
well, this, anywhere, will always fail with bad instruction:
```
print(Thread.current == .main)
let block = {
    print("blah")
}
DispatchQueue.main.sync {
    block()
}
```
at least under 4.2

Reply:
> `sync()` is supposed to be a no-op when the target queue is also the source queue
I didn't quite see where that check is in the code

Reply:
@cnotethegr8 

I tested this, and `DispatchQueue.main.sync { ... }` _does_ crash when called from the main thread.  Elaz's change should be accepted.

Reply:
Great, thanks for the fix @Elaz .